exoscale/datasource_exoscale_compute.go

```diff
@@ -109,112 +109,112 @@ func dataSourceComputeRead(d *schema.ResourceData, meta interface{}) error {
 	computeName, byName := d.GetOk("hostname")
 	computeID, byID := d.GetOk("id")
 	computeTag, byTag := d.GetOk("tags")
+
 	switch {
-	case !byName && !byID && !byTag:
-		return errors.New("either hostname, id or tags must be specified")
-	case computeID != "":
+	case byName:
+		req.Name = computeName.(string)
+
+	case byID:
 		var err error
 		if req.ID, err = egoscale.ParseUUID(computeID.(string)); err != nil {
 			return fmt.Errorf("invalid value for id: %s", err)
 		}
+
 	case byTag:
 		for key, value := range computeTag.(map[string]interface{}) {
 			req.Tags = append(req.Tags, egoscale.ResourceTag{
 				Key:   key,
 				Value: value.(string),
 			})
 		}
+
 	default:
-		req.Name = computeName.(string)
+		return errors.New("either hostname, id or tags must be specified")
 	}
 
 	resp, err := client.GetWithContext(ctx, &req)
 	if err != nil {
 		return err
 	}
+	vm := resp.(*egoscale.VirtualMachine)
 
-	c := resp.(*egoscale.VirtualMachine)
-
-	resp, err = client.GetWithContext(ctx, &egoscale.Volume{
-		VirtualMachineID: c.ID,
-		Type:             "ROOT",
-	})
+	// Querying VM NICs separately because the non-default NICs IP addresses
+	// are not returned in the CS listVirtualMachines operation results.
+	resp, err = client.RequestWithContext(ctx, &egoscale.ListNics{VirtualMachineID: vm.ID})
 	if err != nil {
 		return err
 	}
+	vm.Nic = resp.(*egoscale.ListNicsResponse).Nic
 
-	ds := resp.(*egoscale.Volume).Size >> 30
-
-	resp, err = client.RequestWithContext(ctx, &egoscale.ListNics{
-		VirtualMachineID: c.ID,
+	resp, err = client.GetWithContext(ctx, &egoscale.Volume{
+		VirtualMachineID: vm.ID,
+		Type:             "ROOT",
 	})
 	if err != nil {
 		return err
 	}
-	n := resp.(*egoscale.ListNicsResponse).Nic
+	diskSize := resp.(*egoscale.Volume).Size >> 30
 
-	return dataSourceComputeApply(d, c, n, ds)
+	return dataSourceComputeApply(d, vm, diskSize)
 }
 
-func dataSourceComputeApply(d *schema.ResourceData, compute *egoscale.VirtualMachine, nics []egoscale.Nic, diskSize uint64) error {
-	d.SetId(compute.ID.String())
+func dataSourceComputeApply(d *schema.ResourceData, vm *egoscale.VirtualMachine, diskSize uint64) error {
```

Comment:
This should'n be `instance` instead `vm`.

Reply:
Changed in bc85ea3.